### PR TITLE
Support Unicode in JSON input

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: contrib/epee/include/storages/parserse_base_utils.h [hunk 1]
```diff
@@ -162,6 +185,57 @@ namespace misc_utils
...
+                  val.push_back(0xc0 | (dst >> 6));
+                  val.push_back(0x80 | (dst & 0x3f));
+                }
+                else if (dst <= 0xffff)
+                {
+                  val.push_back(0xe0 | (dst >> 12));
+                  val.push_back(0x80 | ((dst >> 6) & 0x3f));
+                  val.push_back(0x80 | (dst & 0x3f));
+                }
+                else if (dst <= 0x10ffff)
```
[the remark this answers is not in the document]
Comment (moneromooo-monero) [reply contrib/epee/include/storages/parserse_base_utils.h]: Do you know how to encode code points >= 0xffff ? All the examples I've fund use \uxxxx with 3 digits.

[the remark this answers is not in the document]
Comment (moneromooo-monero) [reply contrib/epee/include/storages/parserse_base_utils.h]: Oh you mean this encoding is actually UTF-16, not raw code points ?

[the remark this answers is not in the document]
Comment (vtnerd) [reply contrib/epee/include/storages/parserse_base_utils.h]: Yes, the [RFC](https://tools.ietf.org/html/rfc7159.html) says the value is UTF-16 and a peek at [rapidjson](https://github.com/Tencent/rapidjson/blob/129d19ba7f496df5e33658527a7158c79b99c21c/include/rapidjson/reader.h#L1018) shows that at least one major implementation has done it this way.

**Comment by vtnerd** [comment]
Sorry for not re-reviewing this PR earlier, UTF-16 surrogate pairs are still not handled properly. Anything outside of the Basic Multilingual Plane will decode incorrectly (which are less common characters at least).

**Comment by vtnerd** [comment]
I will try to update it with tests later this week.

**Pull request merged**